### PR TITLE
Implement a way to copy document note field from Order

### PR DESCRIPTION
**Pull request by cristinamghita**

### CODE REVIEW

File: backend/de.metas.business/src/main/java/de/metas/doctype/CopyDocumentNote.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	{
+		final CopyDocumentNote type = typesByCode.get(code);
+		if (type == null)
+		{
+			throw new AdempiereException("No " + CopyDocumentNote.class + " found for code: " + code);
+		}
+		return type;
+	}
+
+	private static final ImmutableMap<String, CopyDocumentNote> typesByCode = Maps.uniqueIndex(Arrays.asList(values()), CopyDocumentNote::getCode);
```
Comment (teosarca) [inline backend/de.metas.business/src/main/java/de/metas/doctype/CopyDocumentNote.java:65]: pls use ReferenceListAwareEnums.index(values())

File: backend/de.metas.business/src/main/java/de/metas/inout/impl/InOutBL.java [hunk 2]
```diff
@@ -683,10 +687,23 @@ public void updateDescriptionAndDescriptionBottomFromDocType(@NonNull final I_M_
...
+
 
 		inOut.setDescription(description.translate(adLanguage));
-		inOut.setDescriptionBottom(documentNote.translate(adLanguage));
+
+		final CopyDocumentNote copyDocumentNote = CopyDocumentNote.ofCode(docType.getCopyDocumentNote());
+
+		if (copyDocumentNote.isCopyDocumentNoteFromOrder() && inOut.getC_Order_ID() > 0)
+		{
+			final String descriptionBottom = orderBL.getDescripttionBottomById(OrderId.ofRepoId(inOut.getC_Invoice_ID()));
```
Comment (teosarca) [inline backend/de.metas.business/src/main/java/de/metas/inout/impl/InOutBL.java:698]: `OrderId.ofRepoId(inOut.getC_Invoice_ID())` .... hmmm?

Comment (cristinamghita) [reply backend/de.metas.business/src/main/java/de/metas/inout/impl/InOutBL.java:698]: Misstake!!!

Thx!

File: backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceBL.java [hunk 3]
```diff
@@ -867,17 +872,29 @@ public void updateDescriptionFromDocTypeTargetId(final org.compiere.model.I_C_In
...
+        	final String descriptionBottom = orderBL.getDescripttionBottomById(OrderId.ofRepoId(invoice.getC_Invoice_ID()));
+			invoice.setDescriptionBottom(descriptionBottom);
 		}
 		else
 		{
-			invoice.setDescriptionBottom(defaultDocumentNote);
+
+			final ITranslatableString documentNote = docTypeTrl.getColumnTrl(I_C_DocType.COLUMNNAME_DocumentNote, docType.getDocumentNote());
+
+			if (!Check.isEmpty(documentNote.toString()))
```
Comment (teosarca) [inline backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceBL.java:887]: pls use TranslatableStrings.isEmpty

Comment (teosarca) [reply backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceBL.java:887]: or... u can Check.isEmpty on documentNote.translate(adLanguage)

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments

**Review by teosarca — APPROVED** [review]
lgtm

**Pull request merged**